hugegraph-server/hugegraph-api/src/main/java/org/apache/hugegraph/api/graph/EdgeAPI.java

```diff
@@ -291,6 +291,19 @@ public String list(@Context GraphManager manager,
         HugeGraph g = graph(manager, graph);
 
         GraphTraversal<?, Edge> traversal;
+
+        // Optimize edge query. Return early when the EdgeLabel does not exist.
+        if (label != null && !g.existsEdgeLabel(label)) {
+            try {
+                traversal = g.traversal().E().limit(0);
+                return manager.serializer(g).writeEdges(traversal, page != null);
+            } finally {
+                if (g.tx().isOpen()) {
+                    g.tx().close();
+                }
+            }
+        }
+
```

Comment:
we can revert this code now.
This code is the scenario of query-edges(not adjacent-edges) with non-exist-label , NOTE we don't need to optimize query-vertices/edges with non-exist-label, see schema.SchemaLabel.getLabelId().